server/rpc.go

```diff
@@ -481,6 +481,7 @@ func SubmitSegment(bcast Broadcaster, seg *stream.HLSSegment, nonce uint64) (*ne
 	case *net.TranscodeResult_Error:
 		err = fmt.Errorf(res.Error)
 		glog.Errorf("Transcode failed for segment %v: %v", seg.SeqNo, err)
+		glog.Info("Ensure the keyframe interval is 4 seconds or less")
```

Comment:
This prints the message unconditionally, when it should be checking for the specific "MediaStats Failure" error before printing.

Note that this is more or less exclusive with the other suggestion of doing the check on the orchestrator; it doesn't really need to be done twice. I have a slight preference for doing the check here, but either way will work.

Apologies if it sounded as if both checks had to be done; it's really one or the other.

Reply:
Ah okay I get it now! The bullet points above are options. I misinterpreted that, apologies on my end. I'll do the check here, and print out `Ensure the keyframe interval is 4 seconds or less` only when `err.Error() == "MediaStats Failed"`